sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparisonSuite.scala

```diff
@@ -406,30 +407,62 @@ class UnwrapCastInBinaryComparisonSuite extends PlanTest with ExpressionEvalHelp
   }
 
   test("SPARK-46069: Support unwrap timestamp type to date type") {
-    val tsLit = Literal.create(ts1, TimestampType)
-    val tsNTZLit = Literal.create(ts1, TimestampNTZType)
-    val floorDate = Cast(tsLit, DateType, Some(conf.sessionLocalTimeZone))
-    val floorDateNTZ = Cast(tsNTZLit, DateType, Some(conf.sessionLocalTimeZone))
-    val dateAddOne = DateAdd(floorDate, Literal(1, IntegerType))
-    val dateAddOneNTZ = DateAdd(floorDateNTZ, Literal(1, IntegerType))
-    assertEquivalent(
-      castTimestamp(f7) > tsLit || castTimestampNTZ(f7) > tsNTZLit,
-      f7 > floorDate || f7 > floorDateNTZ)
-    assertEquivalent(
-      castTimestamp(f7) >= tsLit || castTimestampNTZ(f7) >= tsNTZLit,
-      f7 >= dateAddOne || f7 >= dateAddOneNTZ)
-    assertEquivalent(
-      castTimestamp(f7) === tsLit || castTimestampNTZ(f7) === tsNTZLit,
-      f7 === floorDate && f7 === dateAddOne || f7 === floorDateNTZ && f7 === dateAddOneNTZ)
-    assertEquivalent(
-      castTimestamp(f7) <=> tsLit || castTimestampNTZ(f7) <=> tsNTZLit,
-      FalseLiteral || FalseLiteral)
-    assertEquivalent(
-      castTimestamp(f7) < tsLit || castTimestampNTZ(f7) < tsNTZLit,
-      f7 < dateAddOne || f7 < dateAddOneNTZ)
-    assertEquivalent(
-      castTimestamp(f7) <= tsLit || castTimestampNTZ(f7) <= tsNTZLit,
-      f7 <= floorDate || f7 <= floorDateNTZ)
+    def doTest(tsLit: Literal, tsNTZLit: Literal): Unit = {
+      val floorDate = Cast(tsLit, DateType, Some(conf.sessionLocalTimeZone))
+      val floorDateNTZ = Cast(tsNTZLit, DateType, Some(conf.sessionLocalTimeZone))
+      val dateAddOne = DateAdd(floorDate, Literal(1, IntegerType))
+      val dateAddOneNTZ = DateAdd(floorDateNTZ, Literal(1, IntegerType))
+      val isStartOfDay =
```

Comment:
in test, let's pass in `isStartOfDay` as a parameter. Otherwise the test is repeating the actual code and we don't test much.

Reply:
Done